.github/workflows/set-metadata.yml

```diff
@@ -0,0 +1,354 @@
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".url="${{ needs.get-howso-validator-enterprise-py-details.outputs.url }}"')
+          fi
+          if ! [[ -n "${{ needs.get-howso-synthesizer-py-details.outputs.run-id }}" ]]; then
+            # Append run-id, run-type, build-date, build-title, head-sha, url to metadata JSON
+            details=$(echo "$details" | jq '. + {"howso-engine-py": "{}"}')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".run_id="${{ needs.get-howso-synthesizer-py-details.outputs.run-id }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".run_type="${{ needs.get-howso-synthesizer-py-details.outputs.run-type }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".build_date="${{ needs.get-howso-synthesizer-py-details.outputs.build-date }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".build_title="${{ needs.get-howso-synthesizer-py-details.outputs.build-title }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".head_sha="${{ needs.get-howso-synthesizer-py-details.outputs.head-sha }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".url="${{ needs.get-howso-synthesizer-py-details.outputs.url }}"')
+          fi
+
+          echo "upstream-details=$(echo $details)" >> $GITHUB_OUTPUT
+  
+  set-version:
+    needs: ['set-metadata']
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.choose-version.outputs.version }}
+    steps:
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get previous git tag
+        id: previous-tag
+        run: |
+          tag=$(git for-each-ref --sort=-creatordate --count 5 --format="%(refname:short)" "refs/tags/" | grep -E "^[0-9]+\.[0-9]+\.[0-9]+" | head -n 1)
+          echo "Found tag: $tag"
+          echo "tag=$(echo tag)" >> $GITHUB_OUTPUT
+
+      - name: Get next semver from previous tag
+        id: next-semvers
+        uses: WyriHaximus/github-action-next-semvers@v1
+        with:
+          version: ${{ steps.previous-tag.outputs.tag }}
+
+      - name: Set PR version
+        if: needs.set-metadata.outputs.build-type == 'PR'
+        id: set-pr-version
+        run: |
+          PR_NUMBER=${{ github.event.pull_request.number }}
+          PR_ITERATION=${{ github.run_attempt }}.${{ github.run_number }}
+          echo "version=$(echo ${{ steps.next-semvers.outputs.patch }}-alpha+PR.${PR_NUMBER}.${PR_ITERATION})" >> $GITHUB_OUTPUT
+
+      - name: Set Branch version
+        if: needs.set-metadata.outputs.build-type == 'branch'
+        id: set-branch-version
+        run: |
+          BRANCH_ITERATION=${{ github.run_attempt }}.${{ github.run_number }}
+          echo "version=$(echo ${{ steps.next-semvers.outputs.patch }}-alpha+BR.${{ github.ref_name }}.${BRANCH_ITERATION})" >> $GITHUB_OUTPUT
+
+      - name: Make Version PEP-Compliant
```

Comment:
What happens on a release build with this step? Looks like logic only checks for PR then assumes branch otherwise.